### PR TITLE
[CognitiveServices] LUIS Programmatic Spec - Latest changes

### PR DESCRIPTION
**Pull request by pcostantini**
This PR includes the following changes:
- Those operations that returned empty HTTP 200 responses, now return an OperationStatus object
- Batch adding Examples now return a partial success when some errors are detected
- AvailableEndpoints update to use a Dictionary
- Add deleteUtterances parameter to Delete Intent operation
- Minor description updates

-------

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json [hunk 1]
```diff
@@ -4780,6 +5007,28 @@
           "type": "string"
         }
       },
+      "example": {
+        "name": "FlightClass",
+        "children": [ "Economy", "First" ]
+      }
+    },
+    "CompositeModelCreateObject": {
```
Comment (fearthecowboy) [inline specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: `CompositeModelCreateObject` and `CompositeModelUpdateObject` are essentially the same; can we use one model for this (maybe `CompositeModel` or something?

File: specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json [hunk 2]
```diff
@@ -4126,6 +4331,28 @@
         "children": [ "Location::To" ]
       }
     },
+    "CompositeModelUpdateObject": {
+      "description": "A model object containing the properties to update.",
+      "additionalProperties": false,
```
Comment (fearthecowboy) [inline specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: o_O - `additionalProperties` may not be `false` -- you can just delete this.

File: specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json [hunk 3]
```diff
@@ -6069,6 +6302,20 @@
         }
       }
     },
+    "OperationStatus": {
+      "description": "Response of an Operation status.",
+      "type": "object",
+      "properties": {
+        "code": {
```
Comment (fearthecowboy) [inline specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: Is this the same as the HTTP status that gets returned?

File: specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json [hunk 4]
```diff
@@ -6069,6 +6302,20 @@
...
+          "type": "string",
```
Comment (fearthecowboy) [inline specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: Can you make an enum with the expected values? Are these all 'success' conditions?

Comment (pcostantini) [reply specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: We've seen some operation return "FAILED" and "Failed". I'm not sure if using an enum wont cause an issue when reading those values. Can we leave this as string?

Comment (fearthecowboy) [reply specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: you can use  `x-ms-enum` with `modelAsString` : `true` and unknown values are tolerated, and can be expanded without breaking changes later.

Knowing what's available will help people consume the service easier.

``` json
"enum" : [ "Failed", "FAILED", "Success" ], 
"x-ms-enum" : { 
  "modelAsString": "true",
  "name": "OperationStatusType"
}
```

Comment (pcostantini) [reply specification/cognitiveservices/data-plane/LUIS/Programmatic/v2.0/LUIS-Programmatic.json]: Perfect! It generates proper documentation with possible values, and Constants for referencing the values

**Comment by fearthecowboy** [comment]
Do me a quick favor?

`node -e "fs.writeFileSync('./LUIS-Programmatic.json',JSON.stringify(require('./LUIS-Programmatic.json'),null,2))"`

Can you  run that on the swagger file to make sure we've got the right formatting (the enum array probably isn't quite right there).

Past that, there's some weird error in the validator, but not sure what's causing that. We can commit this; we should probably follow up and make sure we're validating when they look at the glitch.

**Review by fearthecowboy — APPROVED** [review]
Weird glitch still in model validator, but the rest looks good.

**Pull request merged**